packages/blocks/src/api/parser/convert-legacy-block.js

```diff
@@ -66,6 +66,9 @@ export function convertLegacyBlockNameAndAttributes( name, attributes ) {
 	if ( name === 'core/post-comment-date' ) {
 		name = 'core/comment-date';
 	}
+	if ( name === 'core/comments-query-loop' ) {
+		name = 'core/comments';
```

Comment:
Let's maybe document that this alone isn't enough:

```suggestion
		name = 'core/comments';
		// Note that we also had to add a deprecation to the block in order to remove the old,
		// generated `wp-block-comments-query-loop` class name.
```